Library/Homebrew/dev-cmd/linkage.rb

```diff
@@ -16,32 +16,30 @@
 
 module Homebrew
   def linkage
-    found_broken_dylibs = false
     ARGV.kegs.each do |keg|
       ohai "Checking #{keg.name} linkage" if ARGV.kegs.size > 1
       result = LinkageChecker.new(keg)
       if ARGV.include?("--test")
         result.display_test_output
+        Homebrew.failed = true if result.broken_dylibs.any? || result.undeclared_deps.any?
```

Comment:
:+1: on the `Homebrew.failed`, but this line is definitely too long. Could you add methods `broken_dylibs?` and `undeclared_deps?` to `LinkageChecker` and let them handle the check?

I also think that we should stop using `any?` as the inverse of `empty?`, because it is not equivalent to `!empty?`. (But that's a different discussion.)